### PR TITLE
improve-crash-output

### PR DESCRIPTION
**Pull request by maidh91**
# Default

image

# With `--log` param

image

### CODE REVIEW

File: lib/crasher.js [hunk 1]
```diff
@@ -33,19 +33,35 @@ function logCrashAndExit (logPath, errorInfo, checkout, stackTrace, err) {
...
+function printAndExit (errorInfo, stackTrace, err) {
+  // Electron does not play well with process.exit, so use app.exit instead
+  const runContext = isBare ? global.Bare : (global.process.versions.electron ? require('electron').app : global.process)
+  const pid = isBare ? Bare.pid : global.process.pid
+  const errInfo = err !== null && typeof err === 'object' ? JSON.stringify(err, 0, 4).slice(1, -2) : ''
+  const errorMsg = `${stackTrace + errInfo}\n\n${errorInfo}`
+
+  console.error(errorMsg)
+
+  if (isBare) {
```
Comment (rafapaezbas) [inline lib/crasher.js:45]: imo, this part should happen outside the function so we dont have to duplicate

File: lib/crasher.js [hunk 2]
```diff
@@ -33,19 +33,35 @@ function logCrashAndExit (logPath, errorInfo, checkout, stackTrace, err) {
...
+
+function setupCrashHandlers (processName, swap, enableLog) {
   const crashlogPath = path.join(swap, `${processName}.crash.log`)
   const runContext = isBare ? global.Bare : global.process
 
   runContext.on('unhandledRejection', (reason) => {
     const stack = reason?.stack || reason || ''
     const errorInfo = `${processName} exiting due to unhandled rejection`
-    logCrashAndExit(crashlogPath, errorInfo, checkout, stack, reason)
+    enableLog ? logCrashAndExit(crashlogPath, errorInfo, checkout, stack, reason) : printAndExit(errorInfo, stack, reason)
```
Comment (rafapaezbas) [inline lib/crasher.js:59]: I think we should follow the code style convention and use if-else instead of ?

File: lib/crasher.js [hunk 3]
```diff
@@ -33,19 +33,35 @@ function logCrashAndExit (logPath, errorInfo, checkout, stackTrace, err) {
   }
 }
 
-function setupCrashHandlers (processName, swap) {
+function printAndExit (errorInfo, stackTrace, err) {
+  // Electron does not play well with process.exit, so use app.exit instead
+  const runContext = isBare ? global.Bare : (global.process.versions.electron ? require('electron').app : global.process)
```
Comment (rafapaezbas) [inline lib/crasher.js:38]: some of this definitions are the same as logCrashAndExit, can we dedup?

File: lib/crasher.js (hunk 2 above)
Comment (maidh91) [reply lib/crasher.js]: updated

File: lib/crasher.js (hunk 1 above)
Comment (maidh91) [reply lib/crasher.js:43]: updated

File: lib/crasher.js [hunk 4]
```diff
@@ -25,27 +23,43 @@ function logCrashAndExit (logPath, errorInfo, checkout, stackTrace, err) {
...
+  if (enableLog) {
+    logCrash(logPath, errorInfo, checkout, stack, err)
+  } else {
+    printCrash(errorInfo, stack, err)
+  }
 
   if (isBare) {
     os.kill(pid)
   } else {
+    const runContext = isBare ? global.Bare : (global.process.versions.electron ? require('electron').app : global.process)
```
[the remark this answers is not in the document]
Comment (maidh91) [reply lib/crasher.js]: updated

File: lib/crasher.js (hunk 3 above)
Comment (maidh91) [reply lib/crasher.js]: updated

**Pull request merged**